torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -116,6 +116,41 @@ def _call_function_and_unflatten_output(tx, fn, args, kwargs, ret_vt, ret_treesp
     )
 
 
+# NOTE:
+# Mutation detection: We detect mutations by checking the example_value's _version
+# This approach has a corner case that if torch.compile is wrapped
+# by an inference_mode(True) context manager, the tensor version will not be upated.
```

Comment:
Can we add some tests that cond still errors out in a situation where it should if it is under torch.inference_mode?